flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableMetaStoreFactory.java

```diff
@@ -105,6 +105,22 @@ public void createPartition(LinkedHashMap<String, String> partSpec, Path path) t
 			client.add_partition(partition);
 		}
 
+		@Override
+		public void alterPartition(LinkedHashMap<String, String> partitionSpec, Path partitionPath) throws Exception {
+			Partition partition = client.getPartition(database, tableName, new ArrayList<>(partitionSpec.values()));
+			StorageDescriptor partSD = partition.getSd();
```

Comment:
Can we use the copy of thrift object?

Reply:
I'd rather not. The logic is copied from Hive and it doesn't copy all the fields of the SD.